### PR TITLE
Add more gopls settings, use the correct :group value.

### PR DESCRIPTION
**Pull request by vibhavp**
Adds a bunch of settings from https://github.com/golang/tools/blob/6f4f008689da0001e7e1fc8f5f9dcaba40c3bcd3/gopls/doc/settings.md

### CODE REVIEW

**Comment by yyoncho** [comment]
Looks good to me. I will let @muirdm review that.

File: lsp-go.el [hunk 1]
```diff
@@ -211,13 +226,37 @@ $GOPATH/pkg/mod along with the value of
   :group 'lsp-go
   :package-version '(lsp-mode "7.0.1"))
 
+(defcustom lsp-go-complete-unimported nil
```
[the remark this answers is not in the document]
Comment (muirdm) [reply lsp-go.el:229]: I see now the doc incorrectly says this defaults to `false`. It actually defaults to true: https://github.com/golang/tools/blob/ace63f8701f581914a8f6d46ffe84c0b4adebba8/internal/lsp/source/options.go#L99

File: lsp-go.el [hunk 2]
```diff
@@ -211,13 +226,37 @@ $GOPATH/pkg/mod along with the value of
...
+  "If non-nil, the completion engine is allowed to make suggestions for packages that you do not currently import."
+  :type 'boolean
+  :group 'lsp-go
+  :package-version '(lsp-mode "7.0.2"))
+
+(defcustom lsp-go-matcher "caseInsensitive"
```
[the remark this answers is not in the document]
Comment (muirdm) [reply lsp-go.el:235]: I see now the docs say this defaults to "caseInsensitive", which isn't really correct. If you don't specify this setting, it defaults to "fuzzy": https://github.com/golang/tools/blob/ace63f8701f581914a8f6d46ffe84c0b4adebba8/internal/lsp/source/options.go#L96. If you specify this setting but set it to an unsupported value (e.g. "banana"), _then_ if will default to caseInsenstiive: https://github.com/golang/tools/blob/ace63f8701f581914a8f6d46ffe84c0b4adebba8/internal/lsp/source/options.go#L465

**Comment by vibhavp** [comment]
@muirdm We already had the `codelens` experimental setting, so I figured that I'd complete the list. I understand your concern though, so we can always wait for these settings to be made stable.

**Comment by muirdm** [comment]
I would definitely hold off on adding `lsp-go-disabled-annotations` (and `lsp-go-available-annotation-suppressions`). Adding the gc_details code lens to round out the list seems useful.

The "matcher", "unimported", and "staticcheck" settings are probably okay to add as I can imagine people wanting to change them. Can we indicate that the options are experimental somehow? Maybe just in the doc string?

Another useful setting is the "gopls.analyses" to disable certain static check analyses. While also experimental, I feel like most people will want to disable some checks they find annoying. This is what I have for example:

```
(lsp-register-custom-settings
 `(("gopls.analyses" ,(mapcar (lambda (a) (cons a :json-false)) '(ST1000 ST1003 ST1016 ST1021 SA5011 ST1020)))))
```

**Comment by vibhavp** [comment]
Will redo this PR with a bunch of new options that have lately been added to `gopls`.

**Pull request closed without merging**